### PR TITLE
Orphan role assignment cleanup script

### PR DESCRIPTION
**Pull request by matthchr**
Had been meaning to make something like this for a while.

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains tests
- [ ] this PR contains YAML Samples

### CODE REVIEW

File: scripts/v2/clean-orphan-role-assignments.sh [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+while getopts 'y' flag; do
+  case "${flag}" in
+    y) DELETE=1 ;;
+    *) print_usage
+       exit 1 ;;
+  esac
+done
+
+# This lists all of the role assignments with Owner and no principal - in our case that's role assignments f
+IDS=$(az role assignment list --query "([?(roleDefinitionName == 'Owner') && (principalName == '')].id).join(' ', [*])" | tr -d '"')
```
Comment (super-harsh) [inline scripts/v2/clean-orphan-role-assignments.sh:32]: Maybe worth adding a filter for 'Contributor'?

**Review by super-harsh — APPROVED** [review]

File: scripts/v2/clean-orphan-role-assignments.sh (hunk 1 above)
Comment (matthchr) [reply scripts/v2/clean-orphan-role-assignments.sh]: Fixed this and also fixed at least one leak of roleassignments as well.

**Pull request merged**